src/disco/gui/fd_gui.h

```diff
@@ -165,7 +165,7 @@ struct fd_gui_slot {
 
   int leader_state;
 
-  ulong prior_leader_slot;
+  fd_gui_txn_waterfall_t waterfall_begin[ 1 ];
```

Comment:
I don't love doing this because it bloats memory use (we duplicate every waterfall_t twice, one end and one begin.) Can we keep the prior_leader_slot reference, but for the first slot just set it to ULONG_MAX and assume reference waterfall is all zeroes.

Reply:
Actually nvm, this is like 20 mb.

Reply:
Haha at one point I advocated for this prior_leader_slot approach and you were advocating for storing both begin and end.  I think you made a good point back then that having begin stored makes waterfall more available for validators that only have very few rotations per epoch.  For such validators, the prior leader slot might have been overwritten when they try to look at some leader slot.

Reply:
But I agree this duplication is bad for validators that have frequent leader slots.  Maybe we just do something akin to tile timers.  We assume that we're gonna store 4096 leader slot waterfalls, and we point into that storage.